ci-automation/test_env.sh

```diff
@@ -1,11 +1,23 @@
 #!/usr/bin/env bash
 
-# Values: start|end
+on_exit() {
+    echo "$?"
+		docker rm -f centos-test-env
+    exit
+}
+
 command=$1
 
-if [[ "$command" = "start" ]]; then
-	docker build -t lioramilbaum/ubuntu .
-	docker run -it --privileged=true -v $(pwd)/../:/root/ember-csi lioramilbaum/ubuntu
-elif [[ "$command" = "end" ]]; then
-	docker rm $(docker ps -aq)
+if [[ "$command" = "up" ]]; then
+	docker build -t centos-test-env .
+	docker run -d --privileged -d -e 'container=docker' \
+		-v /sys/fs/cgroup:/sys/fs/cgroup:rw -v $(pwd)/../:/root/ember-csi \
+		--name centos-test-env centos-test-env
+	docker exec -i centos-test-env /root/ember-csi/ci-automation/container_up.sh
+	sudo rm -rf .vagrant/machines
+	trap 'on_exit' SIGTERM SIGINT SIGHUP
```

Comment:
You set the trap at the end of the condition. You need to set it in the beginning.

Reply:
+1